Update elastic agent installation script

graphaelli opened the pull request:
## Summary

* require bash and curl
* support arm/aarch
* support macOS

Skipped addressing the sudo requirement, to be discussed.

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

Commit:
Update elastic agent installation script

* require bash and curl
* support arm/aarch
* support macOS

ogupte reviewed (approved):
this is great, thank you!

gbamparop commented on x-pack/plugins/observability_onboarding/public/assets/standalone_agent_setup.sh:
```diff
@@ -1,11 +1,55 @@
+    arch=arm64
+  fi
+elif [ "${OS}" == "Darwin" ]; then
+  os=macos
```
```suggestion
  os=darwin
```

Commit:
fix macOS os

gbamparop commented on x-pack/plugins/observability_onboarding/public/assets/standalone_agent_setup.sh:
```diff
@@ -1,11 +1,55 @@
+  fail "bash required"
+fi
+
+if [ $# -lt 5 ]; then
```
If the "Auto download config" toggle is disabled, `autoDownloadConfig` is not present. Maybe we can sent `autoDownloadConfig=0` instead?

graphaelli replied:
good point, I was back and forth on this and making it required seems like the best option to me

graphaelli replied:
updated to not require for now so we don't get stuck on this.  If we want it later we can update the instructions in kibana at the same time

Commit:
Merge branch 'main' into standalone-agent-script-update

graphaelli commented on x-pack/plugins/observability_onboarding/public/assets/standalone_agent_setup.sh:
```diff
@@ -1,11 +1,55 @@
+fi
+
+if [ $# -lt 5 ]; then
+  fail "usage: $0 api_key api_endpoint agent_version onboarding_id [autoDownloadConfig=1]"
```
we'll have to update this if `autoDownloadConfig` is always specified

gbamparop commented on x-pack/plugins/observability_onboarding/public/assets/standalone_agent_setup.sh:
```diff
@@ -19,41 +63,40 @@ updateStepProgress() {
 echo "Extracting Elastic Agent"
 updateStepProgress "ea-extract" "loading"
-tar -xzf elastic-agent-${ELASTIC_AGENT_VERSION}-linux-x86_64.tar.gz --checkpoint=.1000
+tar -xzf ${artifact}.tar.gz --checkpoint=.1000
```
I get a `tar: Option --checkpoint=.1000 is not supported` error on Intel x64

graphaelli replied:
removed

gbamparop commented on x-pack/plugins/observability_onboarding/public/assets/standalone_agent_setup.sh:
```diff
@@ -19,41 +63,40 @@ updateStepProgress() {
+cd ${artifact}
 ./elastic-agent install -f
 if [ "$?" -eq 0 ]; then
   echo "Elastic Agent installed"
```
That's not necessarily related to this PR but I get the following, although the agent status check failed cc @ogupte 

```
Elastic Agent has been successfully installed.
Elastic Agent installed
Checking Elastic Agent status
.Error: failed to communicate with Elastic Agent daemon: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial unix /var/run/elastic-agent.sock: connect: no such file or directory"
For help, please see our troubleshooting guide at https://www.elastic.co/guide/en/fleet/8.8/fleet-troubleshooting.html
..
Elastic Agent running
Download and save configuration to /opt/Elastic/Agent/elastic-agent.yml
Done with standalone Elastic Agent setup for custom logs. Make sure to add your configuration to /opt/Elastic/Agent/elastic-agent.yml, then look for streaming logs to arrive in Kibana
```

Commit:
remove checkpoint option from tar

for increased portability

graphaelli commented on x-pack/plugins/observability_onboarding/public/assets/standalone_agent_setup.sh:
```diff
@@ -19,41 +63,40 @@ updateStepProgress() {
+echo "Downloading Elastic Agent archive from ${ELASTIC_AGENT_DOWNLOAD_URL}"
 updateStepProgress "ea-download" "loading"
-ELASTIC_AGENT_DOWNLOAD_URL="https://artifacts.elastic.co/downloads/beats/elastic-agent/elastic-agent-${ELASTIC_AGENT_VERSION}-linux-x86_64.tar.gz"
 curl -L -O $ELASTIC_AGENT_DOWNLOAD_URL --fail --continue-at -
```
This fails if the file was fully downloaded already, which can be likely if the script restarted after being killed after download completes.  Reproducible by just running `curl -O https://artifacts.elastic.co/downloads/beats/elastic-agent/elastic-agent-8.8.0-darwin-aarch64.tar.gz -C - --fail` twice in a row (assuming the first succeeds).  This behavior is surprising to me.

A few options we have:
* do nothing.  users will have to delete downloads before trying again
* conditionally resume download.  bake the size of the expected artifact into the script
* remove `--continue-at` option and always download the full artifact
* add another file on disk indicating successful download and skip downloading if it exists

graphaelli replied:
I checked another webserver to see how it behaves and it's consistent with artifacts.elastic.co - a range request starting at the length of the artifact returns an HTTP 416 there as well, so I guess this should not be surprising.

graphaelli replied:
Mentioned this casually to @axw and he added another option - perform a head request to get the full size first and only continue if the size on disk is smaller.  For the sake of simplicity I've removed the `--continue-at` option for now and we can revisit if desired.

Commit:
do not require autoDownloadConfig option

Commit:
switch to fail()

Commit:
Merge branch 'main' into standalone-agent-script-update

Commit:
always perform full download

for now.  --continue-at fails if the full file was already downloaded,  which can be
common if the script is rerun after being killed after download completes

Commit:
Merge branch 'main' into standalone-agent-script-update
